File: src/EnvironmentMonitor.ReactUi/environment-monitor/src/models/deviceInfo.ts
import type { Device } from "./device";
import type { DeviceAttachment } from "./deviceAttachment";
import type { DeviceAttribute } from "./deviceAttribute";
import type { DeviceContact } from "./deviceContact";
import type { SensorInfo } from "./sensor";

export interface DeviceInfo {
  device: Device;
  onlineSince?: Date;
  rebootedOn?: Date;
  lastMessage?: Date;
  showWarning?: boolean;
  attachments: DeviceAttachment[];
  sensors: SensorInfo[];
  defaultImageGuid: string;
  deviceIdentifier: string;
  isVirtual: boolean;
  attributes: DeviceAttribute[];
  contacts: DeviceContact[];
}

File: src/EnvironmentMonitor.Domain/Models/DeviceInfo.cs
using EnvironmentMonitor.Domain.Entities;
using EnvironmentMonitor.Domain.Models.ReturnModel;

namespace EnvironmentMonitor.Domain.Models
{
    public class DeviceInfo
    {
        public Device Device { get; set; }
        public DateTime? OnlineSince { get; set; }
        public DateTime? RebootedOn { get; set; }
        public DateTime? LastMessage { get; set; }
        public Guid? DefaultImageGuid { get; set; }

        public List<SensorExtended> Sensors { get; set; } = [];
    }
}
